org.alloytools.alloy.core/src/main/java/edu/mit/csail/sdg/translator/A4SolutionWriter.java

```diff
@@ -40,6 +41,8 @@
 /**
  * This helper class contains helper routines for writing an A4Solution object
  * out as an XML file.
+ *
+ * @modified: Nuno Macedo, Eduardo Pessoa // [HASLab] electrum-temporal
```

Comment:
prints full trace instance into XML; each state is added as an XML `instance` element; trace meta-data (length, loop, variable elements) is also printed